### ISSUE TITLE
1.7.1 failed to generate wildcard with base domain as SAN (with Gandiv5 api key)

### ISSUE THREAD

**Issue by ylbeethoven** [containous/traefik#3977]
### Do you want to request a *feature* or report a *bug*?
Bug

### What did you do?
I've got a new domain from Gandi.net using gandiv5 api key to test if traefik can generate a wildcard with base domain as SAN.

### What did you expect to see?
I expect the certificate can be generated successfully.

### What did you see instead?
I can see a DNS record being created, but it seems that treafik failed to verified hence the certificate could not be issued.

Here is the entry from Gandi
`_acme-challenge	TXT	300	"CgZfMkwMxPE8CH4tOWrR6Lgr2rrnKZ0qW7vV_Q398v8"`

### Output of `traefik version`: (_What version of Traefik are you using?_)
docker version 1.7.1

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Here is Traefik.toml file
```toml
debug = true

defaultEntryPoints = ["http","https"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]

[acme]
email = "admin@aufomm.pw"
storage = "acme.json"
entryPoint = "https"
  [acme.dnsChallenge]
  provider = "gandiv5" 
  delayBeforeCheck = 0

[[acme.domains]]
  main = "*.aufomm.pw"
  sans = ["aufomm.pw"]
```

Provider : Google Cloud Instance in Singapore
OS : Debian 9

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
time="2018-10-04T06:29:08Z" level=debug msg="Using DNS Challenge provider: gandiv5"
time="2018-10-04T06:45:12Z" level=debug msg="Serving default cert for request: \"traefik.aufomm.pw\""
time="2018-10-04T06:45:13Z" level=debug msg="Serving default cert for request: \"traefik.aufomm.pw\""
time="2018-10-04T06:45:13Z" level=debug msg="http2: server: error reading preface from client 113.29.30.137:52853: remote error: tls: bad certificate"
time="2018-10-04T06:45:13Z" level=debug msg="http2: server: error reading preface from client 113.29.30.137:34382: remote error: tls: bad certificate"
time="2018-10-04T06:49:31Z" level=error msg="Error obtaining certificate retrying in 387.776384ms"
time="2018-10-04T07:09:53Z" level=error msg="Error obtaining certificate: acme:Error -> One or more domains had a problem:\n[aufomm.pw] Time limit exceeded. Last error: NS ns-231-b.gandi.net. did not return the expected TXT record\n"
time="2018-10-04T07:09:53Z" level=error msg="Unable to obtain ACME certificate for domains \"*.aufomm.pw,aufomm.pw\" : unable to generate a certificate for thedomains [*.aufomm.pw aufomm.pw]: acme: Error -> One or more domains had a problem:\n[aufomm.pw] Time limit exceeded. Last error: NS ns-231-b.gandi.net. did notreturn the expected TXT record\n"
```

**Comment by ylbeethoven** [comment]
I've done another test.

With one of my other domains and Cloudflare api key, I was able to get a wildcard with SAN.

```
time="2018-10-04T07:54:48Z" level=debug msg="Using DNS Challenge provider: cloudflare"
_acme-challenge.aufomm.win _acme-challenge.aufomm.win
_acme-challenge.aufomm.win _acme-challenge.aufomm.win
time="2018-10-04T07:55:09Z" level=debug msg="Certificates obtained for domains [*.aufomm.win aufomm.win]"
time="2018-10-04T07:55:09Z" level=debug msg="Configuration received from provider ACME: {}"
time="2018-10-04T07:55:09Z" level=debug msg="Wiring frontend frontend-Host-dash-aufomm-win-1 to entryPoint http"
time="2018-10-04T07:55:09Z" level=debug msg="Creating backend backend-reverse-proxy"
time="2018-10-04T07:55:09Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-dash-aufomm-win-1"
time="2018-10-04T07:55:09Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T07:55:09Z" level=debug msg="Creating server server-traefik-de31cf11f99b781588cbe268a5e5a444 at http://192.168.240.3:8080 with weight 1"
time="2018-10-04T07:55:09Z" level=debug msg="Creating route route-frontend-Host-dash-aufomm-win-1 Host:dash.aufomm.win"
time="2018-10-04T07:55:09Z" level=debug msg="Wiring frontend frontend-Host-dash-aufomm-win-1 to entryPoint https"
time="2018-10-04T07:55:09Z" level=debug msg="Creating backend backend-reverse-proxy"
time="2018-10-04T07:55:09Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-dash-aufomm-win-1"
time="2018-10-04T07:55:09Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T07:55:09Z" level=debug msg="Creating server server-traefik-de31cf11f99b781588cbe268a5e5a444 at http://192.168.240.3:8080 with weight 1"
time="2018-10-04T07:55:09Z" level=debug msg="Creating route route-frontend-Host-dash-aufomm-win-1 Host:dash.aufomm.win"
time="2018-10-04T07:55:09Z" level=debug msg="Wiring frontend frontend-Host-me-aufomm-win-0 to entryPoint http"
time="2018-10-04T07:55:09Z" level=debug msg="Creating backend backend-whoami"
time="2018-10-04T07:55:09Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-me-aufomm-win-0"
time="2018-10-04T07:55:09Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T07:55:09Z" level=debug msg="Creating server server-docker-whoami-1-39dd7884f78dfe4291b32e592deb73fd at http://192.168.240.2:80 with weight 1"
time="2018-10-04T07:55:09Z" level=debug msg="Creating route route-frontend-Host-me-aufomm-win-0 Host:me.aufomm.win"
time="2018-10-04T07:55:09Z" level=debug msg="Wiring frontend frontend-Host-me-aufomm-win-0 to entryPoint https"
time="2018-10-04T07:55:09Z" level=debug msg="Creating backend backend-whoami"
time="2018-10-04T07:55:09Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-me-aufomm-win-0"
time="2018-10-04T07:55:09Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T07:55:09Z" level=debug msg="Creating server server-docker-whoami-1-39dd7884f78dfe4291b32e592deb73fd at http://192.168.240.2:80 with weight 1"
time="2018-10-04T07:55:09Z" level=debug msg="Creating route route-frontend-Host-me-aufomm-win-0 Host:me.aufomm.win"
time="2018-10-04T07:55:09Z" level=debug msg="Add certificate for domains *.aufomm.win,aufomm.win"
time="2018-10-04T07:55:09Z" level=info msg="Server configuration reloaded on :80"
time="2018-10-04T07:55:09Z" level=info msg="Server configuration reloaded on :443"
time="2018-10-04T07:55:09Z" level=info msg="Server configuration reloaded on :8080"
```

**Comment by ylbeethoven** [comment]
I've done a few more tests.

For single wildcard, certificate, No issue

```
time="2018-10-04T08:06:24Z" level=debug msg="Using DNS Challenge provider: gandiv5"
time="2018-10-04T08:06:38Z" level=debug msg="Certificates obtained for domains [*.aufomm.pw]"
time="2018-10-04T08:06:38Z" level=debug msg="Configuration received from provider ACME: {}"
time="2018-10-04T08:06:38Z" level=debug msg="Wiring frontend frontend-Host-traefik-aufomm-pw-0 to entryPoint http"
time="2018-10-04T08:06:38Z" level=debug msg="Creating backend backend-reverse-proxy"
time="2018-10-04T08:06:38Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-traefik-aufomm-pw-0"
time="2018-10-04T08:06:38Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T08:06:38Z" level=debug msg="Creating server server-traefik-3659a13d8bed500a652855be5d8f1575 at http://192.168.48.3:8080 with weight 1"
time="2018-10-04T08:06:38Z" level=debug msg="Creating route route-frontend-Host-traefik-aufomm-pw-0 Host:traefik.aufomm.pw"
time="2018-10-04T08:06:38Z" level=debug msg="Wiring frontend frontend-Host-traefik-aufomm-pw-0 to entryPoint https"
time="2018-10-04T08:06:38Z" level=debug msg="Creating backend backend-reverse-proxy"
time="2018-10-04T08:06:38Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-traefik-aufomm-pw-0"
time="2018-10-04T08:06:38Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T08:06:38Z" level=debug msg="Creating server server-traefik-3659a13d8bed500a652855be5d8f1575 at http://192.168.48.3:8080 with weight 1"
time="2018-10-04T08:06:38Z" level=debug msg="Creating route route-frontend-Host-traefik-aufomm-pw-0 Host:traefik.aufomm.pw"
time="2018-10-04T08:06:38Z" level=debug msg="Wiring frontend frontend-Host-whoami-aufomm-pw-1 to entryPoint http"
time="2018-10-04T08:06:38Z" level=debug msg="Creating backend backend-whoami"
time="2018-10-04T08:06:38Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-whoami-aufomm-pw-1"
time="2018-10-04T08:06:38Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T08:06:38Z" level=debug msg="Creating server server-admin-whoami-1-fafb91d89b9dc9bcaaa609cfca6f8854 at http://192.168.48.2:80 with weight 1"
time="2018-10-04T08:06:38Z" level=debug msg="Creating route route-frontend-Host-whoami-aufomm-pw-1 Host:whoami.aufomm.pw"
time="2018-10-04T08:06:38Z" level=debug msg="Wiring frontend frontend-Host-whoami-aufomm-pw-1 to entryPoint https"
time="2018-10-04T08:06:38Z" level=debug msg="Creating backend backend-whoami"
time="2018-10-04T08:06:38Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-whoami-aufomm-pw-1"
time="2018-10-04T08:06:38Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T08:06:38Z" level=debug msg="Creating server server-admin-whoami-1-fafb91d89b9dc9bcaaa609cfca6f8854 at http://192.168.48.2:80 with weight 1"
time="2018-10-04T08:06:38Z" level=debug msg="Creating route route-frontend-Host-whoami-aufomm-pw-1 Host:whoami.aufomm.pw"
time="2018-10-04T08:06:38Z" level=debug msg="Add certificate for domains *.aufomm.pw"
```

For one common name and one SAN on the same certificate, no issue either.
```
time="2018-10-04T08:09:04Z" level=debug msg="Using DNS Challenge provider: gandiv5"
time="2018-10-04T08:09:47Z" level=debug msg="Certificates obtained for domains [traefik.aufomm.pw whoami.aufomm.pw]"
time="2018-10-04T08:09:47Z" level=debug msg="Configuration received from provider ACME: {}"
time="2018-10-04T08:09:47Z" level=debug msg="Wiring frontend frontend-Host-traefik-aufomm-pw-0 to entryPoint http"
time="2018-10-04T08:09:47Z" level=debug msg="Creating backend backend-reverse-proxy"
time="2018-10-04T08:09:47Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-traefik-aufomm-pw-0"
time="2018-10-04T08:09:47Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T08:09:47Z" level=debug msg="Creating server server-traefik-84b143a8a28023fecf925fee774bcc0f at http://192.168.64.3:8080 with weight 1"
time="2018-10-04T08:09:47Z" level=debug msg="Creating route route-frontend-Host-traefik-aufomm-pw-0 Host:traefik.aufomm.pw"
time="2018-10-04T08:09:47Z" level=debug msg="Wiring frontend frontend-Host-traefik-aufomm-pw-0 to entryPoint https"
time="2018-10-04T08:09:47Z" level=debug msg="Creating backend backend-reverse-proxy"
time="2018-10-04T08:09:47Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-traefik-aufomm-pw-0"
time="2018-10-04T08:09:47Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T08:09:47Z" level=debug msg="Creating server server-traefik-84b143a8a28023fecf925fee774bcc0f at http://192.168.64.3:8080 with weight 1"
time="2018-10-04T08:09:47Z" level=debug msg="Creating route route-frontend-Host-traefik-aufomm-pw-0 Host:traefik.aufomm.pw"
time="2018-10-04T08:09:47Z" level=debug msg="Wiring frontend frontend-Host-whoami-aufomm-pw-1 to entryPoint http"
time="2018-10-04T08:09:47Z" level=debug msg="Creating backend backend-whoami"
time="2018-10-04T08:09:47Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-whoami-aufomm-pw-1"
time="2018-10-04T08:09:47Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T08:09:47Z" level=debug msg="Creating server server-admin-whoami-1-3d9e39135bfa7b221546dcd127409a6f at http://192.168.64.2:80 with weight 1"
time="2018-10-04T08:09:47Z" level=debug msg="Creating route route-frontend-Host-whoami-aufomm-pw-1 Host:whoami.aufomm.pw"
time="2018-10-04T08:09:47Z" level=debug msg="Wiring frontend frontend-Host-whoami-aufomm-pw-1 to entryPoint https"
time="2018-10-04T08:09:47Z" level=debug msg="Creating backend backend-whoami"
time="2018-10-04T08:09:47Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-whoami-aufomm-pw-1"
time="2018-10-04T08:09:47Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-04T08:09:47Z" level=debug msg="Creating server server-admin-whoami-1-3d9e39135bfa7b221546dcd127409a6f at http://192.168.64.2:80 with weight 1"
time="2018-10-04T08:09:47Z" level=debug msg="Creating route route-frontend-Host-whoami-aufomm-pw-1 Host:whoami.aufomm.pw"
time="2018-10-04T08:09:47Z" level=debug msg="Add certificate for domains traefik.aufomm.pw,whoami.aufomm.pw"
```

**Comment by arckosfr** [comment]
Hello,
I'm facing problem with gandiv5 too.
But i don't have the same issue.
```
time="2018-10-04T08:43:03Z" level=debug msg="Building ACME client..."
time="2018-10-04T08:43:03Z" level=debug msg="https://acme-staging-v02.api.letsencrypt.org/directory"
time="2018-10-04T08:43:04Z" level=info msg=Register...
time="2018-10-04T08:43:04Z" level=debug msg="Using DNS Challenge provider: gandiv5"
time="2018-10-04T08:43:04Z" level=debug msg="Delaying 10000000000 rather than validating DNS propagation now."
time="2018-10-04T08:43:05Z" level=error msg="Unable to obtain ACME certificate for domains \"rss.lemark.xyz\" detected thanks to rule \"Host:rss.lemark.xyz\" : unable to generate a certificate for the domains [rss.lemark.xyz]: acme: Error -> One or more domains had a problem:\n[rss.lemark.xyz] error presenting token: request failed with HTTP status code 401\n"
time="2018-10-04T08:43:05Z" level=error msg="Unable to obtain ACME certificate for domains \"lemark.xyz\" detected thanks to rule \"Host:lemark.xyz\" : unable to generate a certificate for the domains [lemark.xyz]: acme: Error -> One or more domains had a problem:\n[lemark.xyz] error presenting token: request failed with HTTP status code 401\n"
time="2018-10-04T08:43:06Z" level=error msg="Unable to obtain ACME certificate for domains \"docker.lemark.xyz\" detected thanks to rule \"Host:docker.lemark.xyz\" : unable to generate a certificate for the domains [docker.lemark.xyz]: acme: Error -> One or more domains had a problem:\n[docker.lemark.xyz] error presenting token: request failed with HTTP status code 401\n"
time="2018-10-04T08:43:07Z" level=error msg="Unable to obtain ACME certificate for domains \"traefik.lemark.xyz\" detected thanks to rule \"Host:traefik.lemark.xyz\" : unable to generate a certificate for the domains [traefik.lemark.xyz]: acme: Error -> One or more domains had a problem:\n[traefik.lemark.xyz] error presenting token: request failed with HTTP status code 401\n"
```
it happen when i want to use dns challenge, san, unique or wildcard
The api key is correct (even regenerate new one)

**Comment by ldez** [comment]
@dominikschubert I made a fix for namecheap, could you try this Docker image [ldez/traefik:namecheap](https://hub.docker.com/r/ldez/traefik/tags/).

**Comment by dominikschubert** [comment]
@ldez Looks good! Wildcard + SAN is working now when using your image. Took about 38s to obtain the certs.

**Comment by nmengin** [comment]
@ylbeethoven 

The fix is made in lego https://github.com/xenolf/lego/pull/664.

We are going to update the dependency ASAP to allow you testing with Træfik.

**Comment by ylbeethoven** [comment]
> @ylbeethoven
> 
> The fix is made in lego [xenolf/lego#664](https://github.com/xenolf/lego/pull/664).
> 
> We are going to update the dependency ASAP to allow you testing with Træfik.

Thanks for getting me back so quickly. I look forward to testing the new version.

By the way, I notice that you do not have Namesilo added to your [list](https://docs.traefik.io/configuration/acme/) yet. 

I've got another domain with Namesilo, if you think it is appropriate, I would love to test their API with Traefik as well. 

Let me know if you are interested and what information you need me to provide.

**Comment by nmengin** [comment]
Many thnks for your prosition @ylbeethoven but `Namesilo` does not seem to be integrated yet in LEGO: https://github.com/xenolf/lego/issues/650

**Comment by traefiker** [comment]
Closed by #3998.
